core/templates/pages/landing-pages/stewards-landing-page/stewards-landing-page.component.ts

```diff
@@ -16,133 +16,148 @@
+        href: 'https://www.oppiafoundation.org/volunteer'
+      }];
+    } else {
+      throw new Error('Invalid tab name: ' + tabName);
+    }
+  }
 
-      ctrl.getStaticSubjectImageUrl = function(subjectName) {
-        return UrlInterpolationService.getStaticImageUrl(
-          '/subjects/' + subjectName + '.svg');
-      };
+  getStaticImageUrl(imagePath: string): string {
+    return this.urlInterpolationService.getStaticImageUrl(imagePath);
+  }
 
-      ctrl.onClickButton = function(buttonDefinition) {
-        SiteAnalyticsService.registerStewardsLandingPageEvent(
-          ctrl.activeTabName, buttonDefinition.text);
-        $timeout(function() {
-          WindowRef.nativeWindow.location = buttonDefinition.href;
-        }, 150);
-      };
+  getStaticSubjectImageUrl(subjectName: string): string {
+    return this.urlInterpolationService.getStaticImageUrl(
+      '/subjects/' + subjectName + '.svg');
+  }
 
-      // Note: This should be kept in sync with the CSS media query on
-      // stewards-landing-page.mainpage.html.
-      var isWindowNarrow = function(windowWidthPx) {
-        return windowWidthPx <= 890;
-      };
+  onClickButton(buttonDefinition: ButtonDefinition): void {
+    this.siteAnalyticsService.registerStewardsLandingPageEvent(
+      this.activeTabName, buttonDefinition.text);
+    setTimeout(() => {
+      this.windowRef.nativeWindow.location.href = buttonDefinition.href;
+    }, 150);
+  }
 
-      ctrl.$onInit = function() {
-        ctrl.TAB_NAME_PARENTS = 'Parents';
-        ctrl.TAB_NAME_TEACHERS = 'Teachers';
-        ctrl.TAB_NAME_NONPROFITS = 'NGOs';
-        ctrl.TAB_NAME_VOLUNTEERS = 'Volunteers';
-        // Set the initial tab name based on the URL; default to
-        // TAB_NAME_PARENTS.
-        var initialPathname = UrlService.getPathname();
-        ctrl.activeTabName = ctrl.TAB_NAME_PARENTS;
-        var URL_PATTERNS_TO_TAB_NAMES = {
-          '/parents': ctrl.TAB_NAME_PARENTS,
-          '/teachers': ctrl.TAB_NAME_TEACHERS,
-          '/partners': ctrl.TAB_NAME_NONPROFITS,
-          '/nonprofits': ctrl.TAB_NAME_NONPROFITS,
-          '/volunteers': ctrl.TAB_NAME_VOLUNTEERS
-        };
-        for (var urlPattern in URL_PATTERNS_TO_TAB_NAMES) {
-          if (initialPathname.indexOf(urlPattern) === 0) {
-            ctrl.activeTabName = URL_PATTERNS_TO_TAB_NAMES[urlPattern];
-            break;
-          }
-        }
-        ctrl.buttonDefinitions = getButtonDefinitions(ctrl.activeTabName);
-        ctrl.windowIsNarrow = isWindowNarrow(
-          WindowDimensionsService.getWidth());
-        ctrl.resizeSubscription = WindowDimensionsService.getResizeEvent().
-          subscribe(evt => {
-            ctrl.windowIsNarrow = isWindowNarrow(
-              WindowDimensionsService.getWidth());
-            $scope.$applyAsync();
-          });
-      };
+  // Note: This should be kept in sync with the CSS media query on
+  // stewards-landing-page.mainpage.html.
```

Comment:
Can you add some test for this?

Reply:
There is no CSS media query on `stewards-landing-page.mainpage.html`.

Reply:
Assuming the media query has been removed and I'm removing this comment.